tools/webpack.config.js

```diff
@@ -7,126 +7,132 @@ const getVersion = require('git-repo-version')
+  // http://webpack.github.io/docs/configuration.html
+  const config = {
+    mode: isDebug ? 'development' : 'production',
+    resolve: {
+      modules: [
+        path.resolve('./node_modules')
+      ]
+    },
+    // The base directory for resolving the entry option
+    context: path.resolve(__dirname, '../src'),
+    // The entry point for the bundle
+    entry: [
+      '!!style-loader!css-loader!normalize.css/normalize.css',
+      /* The main entry point of your JavaScript application */
+      './main.js'
+    ],
+    // Options affecting the output of the compilation
+    output: {
       path: path.resolve(__dirname, '../www/dist'),
-      filename: 'assets.json',
-      prettyPrint: true
-    }),
-    new webpack.LoaderOptionsPlugin({
-      debug: isDebug,
-      minimize: !isDebug
-    })
-  ],
-  // Options affecting the normal modules
-  module: {
-    rules: [
-      {
-        test: /\.jsx?$/,
-        // https://github.com/webpack/webpack/issues/2031#issuecomment-219040479
-        exclude: /node_modules\/(?!(strict-uri-encode)\/).*/,
-        loader: 'babel-loader',
-        options: babelConfig
-      },
-      {
-        test: /\.html$/,
-        use: [{loader: 'html-loader', options: {minimize: true}}]
-      },
-      {
-        test: /\.css$/,
-        include: /node_modules/,
-        loaders: [{loader: 'style-loader'}, {loader: 'css-loader'}]
-      },
-      {
-        test: /\.(png|jpg|jpeg|gif|svg|woff|woff2)$/,
-        use: [
-          {
-            loader: 'url-loader',
-            options: {
-              limit: 10000
-            }
-          },
-          {
-            loader: 'img-loader',
-            options: {
-              enabled: !isDebug,
-              gifsicle: {
-                interlaced: false
-              },
-              mozjpeg: {
-                progressive: true,
-                arithmetic: false
-              },
-              optipng: false,
-              pngquant: {
-                floyd: 0.5,
-                speed: 2
-              },
-              svgo: {
-                plugins: [
-                  {removeTitle: true},
-                  {convertPathData: false}
-                ]
+      publicPath: '/dist/',
+      filename: isDebug ? '[name].js?[hash]' : '[name].[hash].js',
+      chunkFilename: isDebug ? '[id].js?[chunkhash]' : '[id].[chunkhash].js',
+      sourcePrefix: '  '
+    },
+    // Developer tool to enhance debugging, source maps
+    // http://webpack.github.io/docs/configuration.html#devtool
+    devtool: isDebug ? 'source-map' : false,
+    // What information should be printed to the console
+    stats: 'minimal',
+    // The list of plugins for Webpack compiler
+    plugins: [
+      new webpack.DefinePlugin({
+        'process.env.NODE_ENV': isDebug ? '"development"' : '"production"',
+        __DEV__: isDebug,
+        __VERSION__: JSON.stringify(getVersion()),
+        __CONFIG__: JSON.stringify(appConfig)
```

Comment:
Everytime you need a config parameter you use `__CONFIG__` right now in the code.
This means the same string (in this case `JSON.stringify(appConfig)`) is used over and over.

I think it would make sense to create a config.js which exports `__CONFIG__` once as default.

At least that is how I think that the DefinePlugin works.

Reply:
Done